src/app/zap-templates/templates/app/attribute-size-src.zapt

```diff
@@ -0,0 +1,139 @@
+{{> header}}
+
+#include <app/util/af.h>
+#include <app/util/basic-types.h>
+#include <support/SafeInt.h>
+#include <support/logging/CHIPLogging.h>
+
+#include "gen/af-structs.h"
+
+using namespace chip;
+
+// The first 2 bytes specify the number of entries. A value of 0xFFFF means the list in invalid
+// and data is undefined.
+constexpr uint16_t kSizeLengthInBytes = 2u;
+
+void copyListMember(uint8_t * dest, uint8_t * src, bool write, uint16_t * offset, uint16_t length)
+{
+    if (write)
+    {
+        memmove(dest + *offset, src, length);
+    }
+    else
+    {
+        memmove(dest, src + *offset, length);
+    }
+
+    *offset = static_cast<uint16_t>(*offset + length);
+}
+
+uint16_t emberAfCopyList(ClusterId clusterId, EmberAfAttributeMetadata * am, bool write, uint8_t * dest, uint8_t * src, int32_t index)
+{
+    if (index == -1)
+    {
+        memmove(dest, src, am->size);
+        return am->size;
+    }
+
+    if (index == 0)
+    {
+       uint16_t entryCount = emberAfGetInt16u(src, 0, kSizeLengthInBytes);
+       emberAfCopyInt16u(dest, 0, entryCount);
```

Comment:
No, this still does not make sense to me.

If we're trying to read the list length, this will read the first two bytes out `src`, treating them as a little-endian representation of a 16-bit length.  Then it will take that 16-bit value and write it to `dest, in little-endian representation.

The upshot is that we are just doing a 2-byte memcpy from src to dest.

We actually want to have different conceptual behavior depending on whether `write` is true or not, like so:
```
if (write) {
  // src is a pointer to native-endian uint16_t, dst is pointer to buffer that should hold little-endian value
  emberAfCopyInt16u(dst, 0, *reinterpret_cast<uint16_t*>(src));
} else {
  // src is pointer to buffer holding little-endian value, dst is a pointer to native-endian uint16_t
  *reinterpret_cast<uint16_t*>(dst) = emberAfGetInt16u(src, 0, kSizeLengthInBytes);
}
```
or something like that.  We might be able to cheat and claim that endian-swapping is its own inverse so either line will work for what we want no matter what `write` is, as long as all the alignment requirements are satisfied.  And I'm not 100% sure they are (i.e. that on write `dst` is guaranteed to be 0 mod 2, and same for `src` if `!write`).

Reply:
I used the code you suggests.